tests/test_header_matching.js

```diff
@@ -576,3 +576,49 @@ test('multiple interceptors override headers from unrelated request', async t =>
   })
   t.is(res2.statusCode, 200)
 })
+
+// The next three tests cover the special case for the Host header where it's only used for
+// matching if it's defined on the scope and the request. See https://github.com/nock/nock/pull/196
+test('Host header is used for matching if defined on the scope and request', async t => {
+  const scope = nock('http://example.test', {
+    reqheaders: { host: 'example.test' },
+  })
+    .get('/')
+    .reply()
+
+  const { statusCode } = await got('http://example.test/', {
+    headers: { Host: 'example.test' },
+  })
+
+  t.is(statusCode, 200)
+  scope.done()
+})
+
+test('Host header is ignored during matching if not defined on the request', async t => {
+  const scope = nock('http://example.test', {
+    reqheaders: { host: 'example.test' },
```

Comment:
Would changing `example.test` to some other value like `this-is-definitely-not-the-hostname.test` make for a better test?

Reply:
Changed with 3fe496c